targets/TARGET_TOSHIBA/TARGET_TMPM066/us_ticker.c

```diff
@@ -68,47 +74,42 @@ uint32_t us_ticker_read(void)
         us_ticker_init();
     }
 
-    uint32_t tickerbefore = 0;
-    do {
-        tickerbefore = us_ticker;
-        ret_val = (us_ticker * 100);
-    } while (tickerbefore != us_ticker);
+    ret_val = (uint32_t)TMRB_GetUpCntValue(TSB_TB7);
 
     return ret_val;
 }
 
 void us_ticker_set_interrupt(timestamp_t timestamp)
 {
-    uint32_t delta = 0;
-
-    // Stops and clear count operation
-    TSB_T16A1->RUN = TMR16A_STOP;
-    TSB_T16A1->CR = TMR16A_SYSCK;
-    // Set the compare register
-    delta = (timestamp - us_ticker_read());
-    TSB_T16A1->RG = delta;
-    // Set Interrupt
-    NVIC_EnableIRQ(INT16A1_IRQn);
-    // Start TMR_TA1 timer
-    TSB_T16A1->RUN = TMR16A_RUN;
+    NVIC_DisableIRQ(INTTB7_IRQn);
+    NVIC_ClearPendingIRQ(INTTB7_IRQn);
+    TMRB_ChangeTrailingTiming(TSB_TB7, timestamp);
+    //Mask all Interrupts except trailing edge interrupt
+    TMRB_SetINTMask(TSB_TB7, TMRB_MASK_MATCH_LEADINGTIMING_INT | TMRB_MASK_OVERFLOW_INT);
+    NVIC_EnableIRQ(INTTB7_IRQn);
 }
 
 void us_ticker_fire_interrupt(void)
 {
-    NVIC_SetPendingIRQ(INT16A1_IRQn);
+    NVIC_SetPendingIRQ(INTTB7_IRQn);
+    NVIC_EnableIRQ(INTTB7_IRQn);
 }
 
 void us_ticker_disable_interrupt(void)
 {
-    NVIC_DisableIRQ(INT16A1_IRQn);
+    // Mask All interrupts
+    TMRB_SetINTMask(TSB_TB7, TMRB_MASK_MATCH_LEADINGTIMING_INT | TMRB_MASK_MATCH_TRAILINGTIMING_INT | TMRB_MASK_OVERFLOW_INT);
+    // Also clear and disable interrupts by NVIC
+    NVIC_ClearPendingIRQ(INTTB7_IRQn);
+    NVIC_DisableIRQ(INTTB7_IRQn);
 }
 
 void us_ticker_clear_interrupt(void)
 {
-    //no flags to clear
+    INTIFSD_ClearINTReq(INTIFSD_INT_SRC_TMRB_7_MDOVF);
+    NVIC_ClearPendingIRQ(INTTB7_IRQn);
 }
 
 void us_ticker_free(void)
 {
```

Comment:
should this function still be empty?

Reply:
Used only function body for compilation success. However implemented the function now.